Extensions/ISearchContextExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using System;
+using OpenQA.Selenium.Interactions;
+
+
+namespace Extensions
+{
+    public static class ISearchContextExtensions
+    {
+        public static IWebElement WaitAndFindElement(this ISearchContext iSearchContext, By by)
+        {
+            try
+            {
+                var wait = new DefaultWait<ISearchContext>(iSearchContext);
+                wait.Timeout = TimeSpan.FromSeconds(15);
+                return wait.Until(ctx =>
+                {
+                    var elem = ctx.FindElement(by);
+                    wait.IgnoreExceptionTypes(typeof(NoSuchElementException), typeof(StaleElementReferenceException));
+                    if (!elem.Enabled || !elem.Displayed)
+                    {
+                        return null;
+                    }
+
+                    return elem;
+                });
+            }
+            catch (WebDriverTimeoutException)
+            {
+                return null;
+            }
+        }
+
+        public static void WaitUntilElementIs(this ISearchContext iSearchContext, By by, string expectedTextValue)
+        {
+            var wait = new DefaultWait<ISearchContext>(iSearchContext);
+            wait.Timeout = TimeSpan.FromSeconds(20);
+            wait.Until(ctx =>
+            {
+                var elem = ctx.FindElement(by);
+                if (!elem.Enabled || !elem.Displayed)
+                    return null;
+                if (elem.GetAttribute("value") == expectedTextValue)
+                    return elem;
+
+                return null;
```

Comment:
instead of three returns you could do 
```suggestion
                var elem = ctx.FindElement(by);
                
                if (elem.GetAttribute("value") == expectedTextValue &&
                    elem.Enabled &&
                    elem.Displayed)
                {
                    return elem;
                }

                return null;
```